src/main/java/chatengine/ChatEngine.java

```diff
@@ -0,0 +1,168 @@
+                    handleMarkUnmark(parts, command);
+                    break;
+                case LIST:
+                    ioHandler.writeOutput(taskList.displayTasks());
+                    break;
+                case TODO:
+                    handleTodo(parts);
+                    break;
+                case DEADLINE:
+                    handleDeadline(parts);
+                    break;
+                case EVENT:
+                    handleEvent(parts);
+                    break;
+                case DELETE:
+                    handleDelete(parts);
+                    break;
+                case BYE:
+                    return false;
+                case UNKNOWN:
+                default:
+                    throw new ChadException.InvalidCommandException("Unknown command: " + parts[0]);
+            }
+        } catch (NumberFormatException e) {
+            throw new ChadException.InvalidFormatException("Invalid number format.");
+        }
+        return true;
+    }
+
+    private void handleMarkUnmark(String[] parts, CommandType command) throws ChadException {
+        if (parts.length < 2) {
+            throw new ChadException.InvalidArgumentException("Missing index for " + command);
+        }
+        int index = Integer.parseInt(parts[1]) - 1;
+        String response = (command == CommandType.MARK) ? taskList.markTaskAsDone(index) : taskList.markTaskAsNotDone(index);
+        ioHandler.writeOutput(response);
+        saveTasks();
+    }
+
+    private void handleTodo(String[] parts) throws ChadException {
+        if (parts.length < 2) {
+            throw new ChadException.InvalidArgumentException("Missing task description for ToDo.");
+        }
+        taskList.addTodo(parts[1]);
+        ioHandler.writeOutput("Added new ToDo: " + parts[1]);
+        saveTasks();
+    }
+
+    private void handleDeadline(String[] parts) throws ChadException {
+        if (parts.length < 2 || !parts[1].contains(" /by ")) {
+            throw new ChadException.InvalidArgumentException("Invalid format for Deadline. Use: deadline {task} /by {date}");
+        }
+        String[] deadlineParts = parts[1].split(" /by ", 2);
+        try {
+            LocalDateTime dueDate = LocalDateTime.parse(deadlineParts[1], DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+            taskList.addDeadline(deadlineParts[0], dueDate);
+            ioHandler.writeOutput("Added new Deadline: " + deadlineParts[0] + " by " + dueDate);
+            saveTasks();
+        } catch (DateTimeParseException e) {
+            throw new ChadException.InvalidFormatException("Invalid date format. Please use yyyy-MM-dd HH:mm.");
+        }
+    }
+
+    private void handleEvent(String[] parts) throws ChadException {
+        if (parts.length < 2 || !parts[1].contains(" /from ") || !parts[1].contains(" /to ")) {
+            throw new ChadException.InvalidArgumentException("Invalid format for Event. Use: event {task} /from {start} /to {end}");
```

Comment:
Perhaps you can consider using line wrapping if the line length exceeds the limit.